internal/config/common.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/BurntSushi/toml"
+)
+
+func PortToAddress(port int) string {
+	return fmt.Sprintf(":%d", port)
+}
+
+func GenerateAddress(hostname string, port int) string {
+	return fmt.Sprintf("http://%s%s", hostname, PortToAddress(port))
+}
+
+func IsFileEmpty(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.Size() == 0, nil
+}
+
+func GetConfiguration[T any](filename string, defaultGenerator func() T) (T, error) {
+	exePath, err := os.Executable()
+
+	if err != nil {
+		return defaultGenerator(), err
+	}
+
+	exeDir := filepath.Dir(exePath)
+	configPath := filepath.Join(exeDir, filename)
```

Comment:
This seems like a bad way to find the config file. Eg `go run ./cmd/satalite` will have the executable inside `/tmp/go-build/...`, and we don't want to need to place an config file their

Reply:
We absolutely shouldn't be running the binary in production via go run. This is more of a production oriented decision as I'd expect the binary and the configuration file to be localised.

Reply:
Yeah, but it means you're breaking `go run` for local dev, which super sucks, and I don't want to approve code that does that

Reply:
Hmmm, I'll have to think about an elegant way to fix this.